controller/search.go

```diff
@@ -243,3 +247,47 @@ func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
 
 	return ctx.OK(&response)
 }
+
+// Iterate over the WI list and read parent IDs
+// Fetch and load Parent WI in the included list
+func (c *SearchController) enrichWorkItemList(ctx *app.ShowSearchContext, res *app.SearchWorkItemList) {
+	// Need a map of string to UUID to prevent duplicates in place.
+	visitedIDs := map[string]uuid.UUID{}
+	fetchInBatch := []uuid.UUID{}
+	for _, wi := range res.Data {
+		cond := wi.Relationships != nil && wi.Relationships.Parent != nil && wi.Relationships.Parent.Data != nil
+		if cond {
+			parentIDStr := *wi.Relationships.Parent.Data.ID
+			if _, exist := visitedIDs[parentIDStr]; !exist {
+				id, err := uuid.FromString(parentIDStr)
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{
+						"wi_id": parentIDStr,
+						"err":   err,
+					}, "Parent ID is invalid UUID: %s", parentIDStr)
+				}
+				visitedIDs[parentIDStr] = id            // this helps to keep elemtns distinct
```

Comment:
you could probably not use an array here, but only after, by copying all map values in a new array that would be passed to the `appl.WorkItems().LoadBatchByID()` method.

Reply:
also, shouldn't we have the logic of checking for unique values within the `appl.WorkItems().LoadBatchByID()` method ? or even simpler, use the `DISTINCT` keyword in the SQL statement, so you wouldn't need to remove duplicate IDs here.

Reply:
great! will try something like that. thanks @xcoulon

Reply:
@xcoulon Found that psql is helping us with doing `Distinct` implicit on `IN` query.
example:- http://sqlfiddle.com/#!9/f4234d/2/0
I am not aware for how long it will be live. Hence you can try out following
```
CREATE TABLE students (
    id     integer PRIMARY KEY,
    name    varchar(40)
);
INSERT INTO students (id, name) VALUES (1, 'Xavier');
INSERT INTO students (id, name) VALUES (2, 'Baiju');
INSERT INTO students (id, name) VALUES (3, 'Pranav');
select * from students where id IN (1, 2, 3, 1, 2, 3, 4);
``` 

I am adding a test for the same, and removing unwanted code for filtering unique IDs etc in loading WI in batch.

Reply:
ok, great !